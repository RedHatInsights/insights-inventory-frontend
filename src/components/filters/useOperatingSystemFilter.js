import { useState } from 'react';
import { buildOSFilterConfig, buildOSChip } from '../../Utilities/OperatingSystemFilterHelpers';

export const operatingSystemFilterState = { operatingSystemFilter: [] };
export const OPERATING_SYSTEM_FILTER = 'OPERATING_SYSTEM_FILTER';
export const operatingSystemFilterReducer = (_state, { type, payload }) => ({
    ...type === OPERATING_SYSTEM_FILTER && {
        operatingSystemFilter: payload
    }
});

export const useOperatingSystemFilter = ([state, dispatch] = [operatingSystemFilterState]) => {
    const [operatingSystemStateValue, setStateValue] = useState([]);
    const operatingSystemValue = dispatch ? state.operatingSystemFilter : operatingSystemStateValue;
    const setValue = dispatch ? (newValue) => dispatch({ type: OPERATING_SYSTEM_FILTER, payload: newValue }) : setStateValue;
    const filter = buildOSFilterConfig({
        value: operatingSystemValue,
        onChange: (_e, value) => setValue(value)
    });
    const chip = buildOSChip(operatingSystemValue);

    return [filter, chip, operatingSystemValue, setValue];
};
