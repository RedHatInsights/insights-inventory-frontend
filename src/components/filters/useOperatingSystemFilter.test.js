import { renderHook, act } from '@testing-library/react-hooks';
import { useOperatingSystemFilter } from './useOperatingSystemFilter';

describe('useOperatingSystemFilter', () => {
    it('should create filter', () => {
        const { result } = renderHook(() => useOperatingSystemFilter());

        expect(result.current[0]).toMatchSnapshot();
    });

    it('should create chip', () => {
        const { result } = renderHook(() => useOperatingSystemFilter());
        expect(result.current[1]).toMatchSnapshot();

        act(() => {
            result.current[3]({ '8.0': { 8.4: true } });
        });

        expect(result.current[1]).toMatchSnapshot();
    });
});
