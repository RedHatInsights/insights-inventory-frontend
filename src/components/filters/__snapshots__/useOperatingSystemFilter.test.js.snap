// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useOperatingSystemFilter should create chip 1`] = `Array []`;

exports[`useOperatingSystemFilter should create chip 2`] = `
Array [
  Object {
    "category": "Operating System",
    "chips": Array [
      Object {
        "name": "RHEL 8.4",
        "value": "8.4",
      },
    ],
    "type": "operating_system",
  },
]
`;

exports[`useOperatingSystemFilter should create filter 1`] = `
Object {
  "filterValues": Object {
    "groups": Array [
      Object {
        "groupSelectable": true,
        "items": Array [],
        "label": "RHEL 9.0",
        "type": "checkbox",
        "value": "9.0",
      },
      Object {
        "groupSelectable": true,
        "items": Array [
          Object {
            "label": "RHEL 8.6",
            "type": "checkbox",
            "value": "8.6",
          },
          Object {
            "label": "RHEL 8.5",
            "type": "checkbox",
            "value": "8.5",
          },
          Object {
            "label": "RHEL 8.4",
            "type": "checkbox",
            "value": "8.4",
          },
          Object {
            "label": "RHEL 8.3",
            "type": "checkbox",
            "value": "8.3",
          },
          Object {
            "label": "RHEL 8.2",
            "type": "checkbox",
            "value": "8.2",
          },
          Object {
            "label": "RHEL 8.1",
            "type": "checkbox",
            "value": "8.1",
          },
        ],
        "label": "RHEL 8.0",
        "type": "checkbox",
        "value": "8.0",
      },
      Object {
        "groupSelectable": true,
        "items": Array [
          Object {
            "label": "RHEL 7.9",
            "type": "checkbox",
            "value": "7.9",
          },
          Object {
            "label": "RHEL 7.8",
            "type": "checkbox",
            "value": "7.8",
          },
          Object {
            "label": "RHEL 7.7",
            "type": "checkbox",
            "value": "7.7",
          },
          Object {
            "label": "RHEL 7.6",
            "type": "checkbox",
            "value": "7.6",
          },
          Object {
            "label": "RHEL 7.5",
            "type": "checkbox",
            "value": "7.5",
          },
          Object {
            "label": "RHEL 7.4",
            "type": "checkbox",
            "value": "7.4",
          },
          Object {
            "label": "RHEL 7.3",
            "type": "checkbox",
            "value": "7.3",
          },
          Object {
            "label": "RHEL 7.2",
            "type": "checkbox",
            "value": "7.2",
          },
          Object {
            "label": "RHEL 7.1",
            "type": "checkbox",
            "value": "7.1",
          },
        ],
        "label": "RHEL 7.0",
        "type": "checkbox",
        "value": "7.0",
      },
      Object {
        "groupSelectable": true,
        "items": Array [
          Object {
            "label": "RHEL 6.10",
            "type": "checkbox",
            "value": "6.10",
          },
          Object {
            "label": "RHEL 6.9",
            "type": "checkbox",
            "value": "6.9",
          },
          Object {
            "label": "RHEL 6.8",
            "type": "checkbox",
            "value": "6.8",
          },
          Object {
            "label": "RHEL 6.7",
            "type": "checkbox",
            "value": "6.7",
          },
          Object {
            "label": "RHEL 6.6",
            "type": "checkbox",
            "value": "6.6",
          },
          Object {
            "label": "RHEL 6.5",
            "type": "checkbox",
            "value": "6.5",
          },
          Object {
            "label": "RHEL 6.4",
            "type": "checkbox",
            "value": "6.4",
          },
          Object {
            "label": "RHEL 6.3",
            "type": "checkbox",
            "value": "6.3",
          },
          Object {
            "label": "RHEL 6.2",
            "type": "checkbox",
            "value": "6.2",
          },
          Object {
            "label": "RHEL 6.1",
            "type": "checkbox",
            "value": "6.1",
          },
        ],
        "label": "RHEL 6.0",
        "type": "checkbox",
        "value": "6.0",
      },
    ],
    "onChange": [Function],
    "selected": Array [],
  },
  "label": "Operating System",
  "onChange": [Function],
  "type": "group",
  "value": Array [],
}
`;
